src/types/traits.rs

```diff
@@ -14,13 +14,27 @@ pub trait AsTypeRef {
 
 macro_rules! trait_type_set {
     ($trait_name:ident: $($args:ident),*) => (
-        pub trait $trait_name: AsTypeRef + Debug {}
-
         $(
             impl $trait_name for $args {}
         )*
     );
 }
 
+/// Represents any LLVM type.
+pub trait AnyType: AsTypeRef + Debug {
+    /// Returns an `AnyTypeEnum` that represents the current type.
+    fn as_any_type_enum(&self) -> AnyTypeEnum {
```

Comment:
It's interesting to note that these basically seem like associated types

Reply:
They kind of do, although not in a generic way